### DISCUSSION TITLE
Annoying and slow performance with golang and autosave enabled

### DISCUSSION THREAD

**Discussion by hareldvd** [pulsar-edit/.github#192] [General]
Hi, 
Using pulsar with golang, whenever save is performed, full compilation is performed including unit testing. 
Having the autosave active saves the buffer on window leave and on every **mouse click**. Because of that, every mouse click, which initiates save, compiles my code and show the errors, this could have been nice unless it would take a lot of time (I guess on better computers it will also be significant).

Any idea?

**Comment by confused-Techie** [comment]
By default Pulsar shouldn't be doing any of this. When Pulsar autosaves, since it's a text editor it only cares about the text. I'd make sure to take a look at any community package's you have installed, and maybe even take a look your `init.js` user config, as maybe there is something in there.

But this absolutely isn't likely happening because of Pulsar itself.

Taking a quick look around at some of the most popular community packages for Golang, it looks like the community package `go-plus` has a setting called "Compile on Save" which according to it's description: "Compile the current package when a file is saved and provide linter errors.".

So this is most likely you're issue. You'll need to navigate to this community package's settings and disable this setting, as well as any others you'd like to

**Reply by hareldvd** [reply]
Thanks for your reply,

You are absolutely right regarding that pulsar does not perform the compilation. The compilation is performed by the docker that runs my go implementation (similar to what npm start does in Angular environment).

My question is if there are other ways to make the save only when I exit the pulsar window and not on any mouse click as it does now.
Thanks.

**Reply by confused-Techie** [reply]
I see. Wasn't clear on what behavior you wanted to change from the original message.

But in that case, I've just taken a look through the source code of `autosave`, and this also shouldn't be saving on 'any mouse click'. The `autosave` package is only able to save when an Editor loses focus, is destroyed, or when the window is closed. That's all it does be default. 

One thing you could do is disable autosave entirely. Then when you try to close Pulsar, you'll be asked if you want to save. While yes it's an additional click, might be the simpliest answer.

Otherwise, what you may be seeing, is the `autosave` package exposes a way to let other community package's inform it if it needs to save.

Meaning that you could have some additional package installed, that's telling autosave plus to save all the time. It looks like there's a few community package's that utilize the ServiceHub to communicate to `autosave`. Take a look to see if you have any [of these packages](https://web.pulsar-edit.dev/packages?service=autosave&serviceType=consumed) installed.

**Reply by hareldvd** [reply]
OK, I figured out that go-plus package: https://web.pulsar-edit.dev/packages/go-plus
Trying to go to the github project I noticed that the project has been archived as many other atom projects have.
Are those bad news?
